code/routers/country_page/country.js

```diff
@@ -99,28 +99,192 @@ router.get('/get-country-info', (req, res) => {
 /*
     Get Country organizations api
     inputs:
-      countryName: String
+      name: String
 
-    Returns the list of organizations associated with given country
+    example query:
+      /api/country-page/country/get-organizations?name=Germany
+
+    Returns the list of organizations associated with country's organizationIDs
 */
 router.get('/get-organizations', (req, res) => {
-  name = req.body.name;
-
-  var query = {name: name};
+  var query = req.query;
   Country.find(query, function(err, country) {
-    if(err) {
+    if(err || country.length == 0) {
       return res.status(countryCodes.getOrganizations.countryNotFound.status).send({
         message: countryCodes.getOrganizations.countryNotFound.message
       });
     }else{
-      return res.status(countryCodes.getOrganizations.success.status).send({
-        organizations: country[0].organizations,
-        message: countryCodes.getOrganizations.success.message
+      var orgs = country[0].organizationIDs;
+      var orgQuery = {_id: {$in: orgs}}; // this queries a list of ids
+      Organization.find(orgQuery, function(err, organizations) {
+        if(err || organizations.length == 0){
+          return res.status(countryCodes.getOrganizations.organizationsNotFound.status).send({
+            message: countryCodes.getOrganizations.organizationsNotFound.message
+          });
+        }else{
+          return res.status(countryCodes.getOrganizations.success.status).send({
+           organizations: organizations,
+           message: countryCodes.getOrganizations.success.message
+          });
+        }
+      })
+    }
+  })
+})
+
+
+/*
+    Insert Organizations API
+    inputs:
+      [{
+      countryName: String,
+      orgName: String
+      }]
+
+    Expects all countryNames to be the same.
+*/
+router.post('/insert-organizations', (req, res) => {
+
+  //Handling of multiple organizations, still expecting them to be from the same country
+  if(Array.isArray(req.body)){
+    var orgNames = [];
+    for (var i = 0; i < req.body.length; i++){
+      var curBody = req.body[i];
+      orgNames.push(curBody.orgName);
+    }
+    var orgQuery = {orgName: {$in: orgNames}};
+    var countryQuery = {name: req.body[0].countryName}; 
+  }else{
+    var orgQuery = {orgName: req.body.orgName}; 
+    var countryQuery = {name: req.body.countryName};  
+  }
+  
+
+  Organization.find(orgQuery, function(err, organizations) {
+    if(err || organizations.length == 0){
+      return res.status(countryCodes.insertOrganizations.organizationsNotFound.status).send({
+        message: countryCodes.insertOrganizations.organizationsNotFound.message
       });
+    }else{
+      var updatedOrgIDs = []
+      Country.find(countryQuery, function(err, country) {
+        if(err) {
```

Comment:
This should check if the length of country is > 0 otherwise line 176 can cause an error